packages/astro/src/core/ssr/index.ts

```diff
@@ -79,7 +79,8 @@ export async function ssr({ astroConfig, filePath, logging, mode, origin, pathna
     // Important: This needs to happen first, in case a renderer provides polyfills.
     const renderers = await resolveRenderers(viteServer, astroConfig);
     // Load the module from the Vite SSR Runtime.
-    const mod = (await viteServer.ssrLoadModule(fileURLToPath(filePath))) as ComponentInstance;
+    const viteFriendlyURL = `/@fs${filePath.pathname}`;
+    const mod = (await viteServer.ssrLoadModule(viteFriendlyURL)) as ComponentInstance;
```

Comment:
This is a bit of a hack, but it turns out Vite uses these URLs internally for Windows. TIL that this doesn’t necessarily have to be a “correct” path on disk; just a URL that Vite understands.